feat: Add release namespace to all resources

peschmae opened the pull request:
#### Overview

This PR adds the `namespace` to all resources, using the `Relase.Namespace` provided by Helm.

#### What this PR does / why we need it

Fixes #120

As mentioned in #120 the Chart currently doesn't set the namespace on the resources, resulting in different behavior for `helm template` and `helm install` commands, as documented in https://github.com/helm/helm/issues/3553

The recommended solution by Helm is to set the namespace explicitly, as is done by a lot of other Charts (eg. cert-manager just to name one).

#### Checklist

- [x] Change log updated in `Chart.yaml` (see the contributing guide for details)
- [x] Chart version bumped in `Chart.yaml` (see the contributing guide for details)
- [x] Documentation regenerated by running `make docs`

husira commented:
Any updates on this Pull Request?

TylerGillson commented:
@sagikazarmark please consider this contribution 🙏🏼

mattwelke commented:
I recommend going with a slightly different approach which makes the chart even more flexible. If each resource uses `{{ .Values.namespace | default .Release.Namespace }}`, it enables users of the chart deploying its resources into a specific namespace. This is helpful for use cases like using this chart as a subchart.

You can see an example of this approach in action in the cert-manager chart: https://github.com/cert-manager/cert-manager/blob/master/deploy/charts/cert-manager/templates/_helpers.tpl#L172

They define a helper there, and then they use the helper for each resource. Example for deployment: https://github.com/cert-manager/cert-manager/blob/master/deploy/charts/cert-manager/templates/deployment.yaml#L5

peschmae commented:
I've rebased the pull request to the latest state of `master` and implemented the proposed change by @mattwelke 

I've opted to use `namespaceOverride` as the values name, to keep it inline with `nameOverride` and `fullnameOverride`.

In the helper I've also linked the same helm issue, as referenced by cert-manager, and I made sure that the namespace conforms the RFC 1123 Label Names as required by Kubernetes (just like it's done for the name override)

peschmae commented:
I've now also added a `dex.serviceMonitor.namespace` helper, which properly uses the `serviceMonitor.namespace` value, and falls back to the regular `dex.namespace` helper otherwise.

The selector in the serviceMonitor now also uses the `dex.namespace` helper

sagikazarmark reviewed (approved):
Thanks @peschmae!

mattwelke commented:
Thanks @peschmae!
